controller/statsyncer/statsyncer.go

```diff
@@ -0,0 +1,243 @@
+package statsyncer
+
+import (
+	"fmt"
+	"time"
+
+	clusterv1 "github.com/rancher/types/apis/cluster.cattle.io/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	syncInterval = 5 * time.Second
+)
+
+type StatSyncer struct {
+	clusterName  string
+	Clusters     clusterv1.ClusterInterface
+	ClusterNodes clusterv1.ClusterNodeInterface
+	Pods         corev1.PodInterface
+	Nodes        corev1.NodeInterface
+}
+
+func Register(workload *config.WorkloadContext) {
+	s := &StatSyncer{
+		clusterName:  workload.ClusterName,
+		Clusters:     workload.Cluster.Cluster.Clusters(""),
+		ClusterNodes: workload.Cluster.Cluster.ClusterNodes(""),
+		Pods:         workload.K8sClient.CoreV1().Pods(""),
+		Nodes:        workload.K8sClient.CoreV1().Nodes(),
+	}
+
+	go s.syncResources(syncInterval)
+}
+
+func (s *StatSyncer) syncResources(syncInterval time.Duration) {
+	for range time.Tick(syncInterval) {
+		err := s.updateClusterNodeResources()
+		logrus.Info("Syncing allocated resources")
+		if err != nil {
+			logrus.Warn(err)
+		}
+	}
+}
+
+func (s *StatSyncer) updateClusterNodeResources() error {
+	cluster, err := s.getCluster()
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Info("Skip syncing node resources, cluster [%s] not found", s.clusterName)
+			return nil
+		}
+		return err
+	}
+	if cluster == nil {
+		logrus.Info("Skip syncing node resources, cluster [%s] deleted", s.clusterName)
+		return nil
+	}
+	if !isProvisioned(cluster) {
+		return fmt.Errorf("Skip syncing node resources - cluster [%s] not provisioned yet", s.clusterName)
+	}
+	nodes, err := s.Nodes.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Skip syncing node resources - Error getting nodes %v", err)
+	}
+	cnodes, err := s.ClusterNodes.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Skip syncing node resources - Error getting cluster nodes %v", err)
+	}
+	nodeNameToNode := s.getNodeMapping(nodes)
+	// ans, _ := json.Marshal(nodeNameToNode)
+	// logrus.Infof("nodes mapping %v", string(ans))
+	for _, cnode := range cnodes.Items {
+		node := nodeNameToNode[cnode.NodeName]
+		if node == nil {
+			logrus.Infof("Skip adding cluster node resources [%s] Error getting Node %v", node.Name, err)
+			continue
+		}
+		pods, err := s.getNonTerminatedPods(node.Name)
+		if err != nil {
+			logrus.Infof(err.Error())
```

Comment:
we should log it in WARN, and may be some custom message like "Can't calculate requested/limit capacity due to failing to list pods`